api/v1beta2/types.go

```diff
@@ -134,12 +134,42 @@ var (
 	ResourceTypeVPC = ResourceType("vpc")
 	// ResourceTypeSubnet is VPC subnet resource.
 	ResourceTypeSubnet = ResourceType("subnet")
+	// ResourceTypePublicGateway is a VPC Public Gateway resource.
+	ResourceTypePublicGateway = ResourceType("publicGateway")
+	// ResourceTypeSecurityGroup is a VPC Security Group resource.
+	ResourceTypeSecurityGroup = ResourceType("securityGroup")
 	// ResourceTypeCOSInstance is IBM COS instance resource.
 	ResourceTypeCOSInstance = ResourceType("cosInstance")
 	// ResourceTypeResourceGroup is IBM Resource Group.
 	ResourceTypeResourceGroup = ResourceType("resourceGroup")
 )
 
+// COSInstanceReference represents an IBM Cloud COS instance.
+type COSInstanceReference struct {
+	// name defines the name of the COS Instance.
+	// +kubebuilder:validation:MinLength:=3
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Pattern=`^[a-z0-9][a-z0-9.-]{1,61}[a-z0-9]$`
+	// +required
+	Name string `json:"name"`
+
+	// id is the ID of the COS instance.
+	// +optional
+	ID *string `json:"id,omitempty"`
+
+	// bucketName is IBM Cloud COS bucket name.
+	// +optional
+	BucketName *string `json:"bucketName,omitempty"`
+
+	// bucketID is the IBM Cloud COS bucket ID.
+	// +optional
+	BucketID *string `json:"bucketID,omitempty"`
```

Comment:
same comment as cos (do we really need both Name and ID field?)

Reply:
Same as above, I can remove if you have found both to be unnecessary.

Reply:
Same update as above, will try only relying on ID's

Reply:
I dropped BucketName for now.